src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -31,6 +31,55 @@ def get_location_completion_list(cmd, prefix, namespace, **kwargs):  # pylint: d
     return [l.name for l in result]
 
 
+# pylint: disable=redefined-builtin
+def get_datetime_type(help=None, date=True, time=True, timezone=True):
+
+    help_string = help + ' ' or ''
+    accepted_formats = []
+    if date:
+        accepted_formats.append('date (yyyy-mm-dd)')
+    if time:
+        accepted_formats.append('time (hh:mm:ss.xxxxx)')
+    if timezone:
+        accepted_formats.append('timezone (+/-hh:mm)')
+    help_string = help_string + 'Format: ' + ', '.join(accepted_formats)
+
+    # pylint: disable=too-few-public-methods
+    class DatetimeAction(argparse.Action):
+
+        def __call__(self, parser, namespace, values, option_string=None):
+            """ Parse a date value and return the ISO8601 string. """
+            import dateutil.parser
+
+            value_string = ' '.join(values)
+            dt_val = None
+            try:
+                # attempt to parse ISO 8601
+                dt_val = dateutil.parser.parse(value_string)
+            except ValueError:
+                pass
+
+            # TODO: custom parsing attempts here
```

Comment:
What's the plan to fill in the TODO here? Open an issue?

Reply:
Issue #5649 is the issue for this.  I will mention it in the comment.